### PR TITLE
fb-YAHOO-215

### PR DESCRIPTION
**Pull request by haghard**

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.hadoop.util.ExitUtil;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class TestYarnUncaughtExceptionHandler {
+
+  private static final YarnUncaughtExceptionHandler exHandler = 
+        new YarnUncaughtExceptionHandler();
+  /**
+   *
```
Comment (aleksgor) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: I suppose there need some comments

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+public class TestYarnUncaughtExceptionHandler {
+
+  private static final YarnUncaughtExceptionHandler exHandler = 
+        new YarnUncaughtExceptionHandler();
+  /**
+   * 
+   * @throws InterruptedException
+   */
+  @Test(timeout = 15000)
+  public void testUncaughtExceptionHandlerWithtRuntimeException()
```
Comment (aleksgor) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: Witht  - misprint ?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    Mockito.verify(spyYarnHandler).uncaughtException(yarnThread, yarnException);
+  }
+
+  /**
+   * used {@code ExitUtil} class to avoid jvm exit through
+   * {@code System.exit(-1) }
+   * 
+   * @throws InterruptedException
+   */
+  @Test(timeout = 15000)
```
Comment (aleksgor) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: This is very short test. May be it is  enough one  sec  ?

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    yarnThread.setUncaughtExceptionHandler(spyYarnHandler);
+    assertSame(spyYarnHandler, yarnThread.getUncaughtExceptionHandler());
+    yarnThread.start();
+    yarnThread.join();
+    Mockito.verify(spyYarnHandler).uncaughtException(yarnThread, yarnException);
+  }
+
+  /**
+   * used {@code ExitUtil} class to avoid jvm exit through
+   * {@code System.exit(-1) }
```
Comment (dennisyv) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: doesn't it produce javadoc warnings?

Comment (dennisyv) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: idea says that minus ("-") is invalid character for javadoc

Comment (haghard) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/test/java/org/apache/hadoop/yarn/TestYarnUncaughtExceptionHandler.java]: no  
{color:green}+1 javadoc{color}.  The javadoc tool did not generate any warning messages.

**Pull request closed without merging**